### DISCUSSION TITLE
need support for create_on_conflict in CreateTableRelation

### DISCUSSION THREAD

**Discussion by scgkiran** [duckdb/duckdb#15232] [Ideas]
Why do you want this feature?

We need this to support CTAS create mode in duckdb-substrait extension
I have [this change](https://github.com/scgkiran/duckdb/pull/1/files) to support this. Please let me know if I can raise a PR for this.

**Comment by Tishj** [comment]
Looks reasonable, this will need accompanying c++ tests to test the changes but other than it looks good to me

**Reply by scgkiran** [reply]
Thank you @Tishj. I raised PR https://github.com/duckdb/duckdb/pull/15245 please take a look.
